src/ring_theory/ideal/operations.lean

```diff
@@ -1174,7 +1212,54 @@ surjective_quot_mk _
 lemma quotient.mkₐ_ker (I : ideal A) : (quotient.mkₐ R I).to_ring_hom.ker = I :=
 ideal.mk_ker
 
-variable {R}
+variables {R} {B : Type*} [comm_ring B] [algebra R B]
+
+lemma quotient.smul (f : A →ₐ[R] B) (r : R) (x : f.to_ring_hom.ker.quotient) :
+  f.to_ring_hom.ker_lift (r • x) = r • f.to_ring_hom.ker_lift x :=
+begin
+  obtain ⟨a, rfl⟩ := quotient.mkₐ_surjective R _ x,
+  rw [← alg_hom.map_smul, quotient.mkₐ_eq_mk, ring_hom.ker_lift_mk],
+  exact f.map_smul _ _
+end
+
+/-- The induced algebras morphism from the quotient by the kernel to the codomain.
+
+This is an isomorphism if `f` has a right inverse (`quotient_ker_alg_equiv_of_right_inverse`) / is surjective (`quotient_ker_alg_equiv_of_surjective`).
+-/
+def ker_lift_alg (f : A →ₐ[R] B) : f.to_ring_hom.ker.quotient →ₐ[R] B :=
+alg_hom.mk' f.to_ring_hom.ker_lift (λ _ _, quotient.smul f _ _)
+
+@[simp]
+lemma ker_lift_alg_mk (f : A →ₐ[R] B) (a : A) :
+  ker_lift_alg f (quotient.mk f.to_ring_hom.ker a) = f a := rfl
+
+@[simp]
+lemma ker_lift_alg_to_ring_hom (f : A →ₐ[R] B) :
+  (ker_lift_alg f).to_ring_hom = ring_hom.ker_lift f := rfl
+
+/-- The induced algebra morphism from the quotient by the kernel is injective. -/
+lemma ker_lift_alg_injective (f : A →ₐ[R] B) : function.injective (ker_lift_alg f) :=
+ring_hom.ker_lift_injective f
+
+/-- The first isomorphism theorem for agebras, computable version. -/
+def quotient_ker_alg_equiv_of_right_inverse
+  {f : A →ₐ[R] B} {g : B → A} (hf : function.right_inverse g f) :
+  f.to_ring_hom.ker.quotient ≃ₐ[R] B :=
+{ to_fun := ker_lift_alg f,
+  inv_fun := (quotient.mkₐ R f.to_ring_hom.ker) ∘ g,
+  left_inv := begin
+    rintro ⟨a⟩,
+    apply ker_lift_alg_injective,
+    change f (g (f a)) = f a,
+    exact hf (f a),
+  end,
+  right_inv := hf,
```

Comment:
Does it work if you replace these with:
```suggestion
{ ..quotient_ker_equiv_of_right_inverse hf,
```

Reply:
It doesn't work:
```
type mismatch at application
  ring_hom.quotient_ker_equiv_of_right_inverse hf
term
  hf
has type
  function.right_inverse g ⇑f
but is expected to have type
  function.right_inverse ?m_3 ⇑?m_6
```
I tried some modifications but I wasn't able to make it working. The function is the same as for rings, but how can Lean guess that it is an algebra homomorphism?

Reply:
Maybe `(\lam x, hf x)` instead of `hf`?

Reply:
```
type mismatch at application
  ring_hom.quotient_ker_equiv_of_right_inverse _
term
  λ (x : B), hf x
has type
  ∀ (x : B), ⇑f (g x) = x
but is expected to have type
  function.right_inverse ?m_2 ⇑?m_4
```
:( But this works
```
def quotient_ker_alg_equiv_of_right_inverse
  {f : A →ₐ[R] B} {g : B → A} (hf : function.right_inverse g f.to_ring_hom) :
  f.to_ring_hom.ker.quotient ≃ₐ[R] B :=
{ ..ring_hom.quotient_ker_equiv_of_right_inverse hf,
  ..ker_lift_alg f}
```
The difference is that now in the assumptions there is `f.to_ring_hom`. The function is the same and `alg_hom.coe_to_ring_hom` seems relevant, but I don't know how to tell Lean to use it.

Reply:
Ah, this is better!
```
/-- The first isomorphism theorem for agebras, computable version. -/
def quotient_ker_alg_equiv_of_right_inverse
  {f : A →ₐ[R] B} {g : B → A} (hf : function.right_inverse g f) :
  f.to_ring_hom.ker.quotient ≃ₐ[R] B :=
{ ..ring_hom.quotient_ker_equiv_of_right_inverse (λ x, show f.to_ring_hom (g x) = x, from hf x),
  ..ker_lift_alg f}
```

Reply:
Excellent, I figured it would be defeq with enough convincing!